### PR TITLE
[Telemetry] Add a dashboard usage

### PR DESCRIPTION
**Pull request by rkooo567**
Signed-off-by: SangBin Cho <rkooo567@gmail.com>

## Why are these changes needed?

When the dashboard is first used, it records the usage

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rickyyx — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/primevideo/">Amazon Prime Video</a> on <a href="https://giphy.com/gifs/primevideo-2020-borat-subsequent-moviefilm-Od0QRnzwRBYmDU3eEO">GIPHY</a></div>

File: python/ray/tests/test_usage_stats.py [hunk 1]
```diff
@@ -1463,6 +1463,45 @@ def verify():
...
+        ray.init(address=cluster.address)
+
+        """
+        Verify the usage_stats.json contains the lib usage.
+        """
+        temp_dir = pathlib.Path(ray._private.worker._global_node.get_session_dir_path())
+        wait_for_condition(lambda: file_exists(temp_dir), timeout=30)
+
+        def verify():
+            dashboard_enalbed = read_file(temp_dir, "usage_stats")["extra_usage_tags"][
```
Comment (ericl) [inline python/ray/tests/test_usage_stats.py:1489]:
```suggestion
            dashboard_enabled = read_file(temp_dir, "usage_stats")["extra_usage_tags"][
```

**Comment by ericl** [comment]
Related test failures in minimal

**Comment by rkooo567** [comment]
@ericl @rickyyx After talking to Alan, we decided to record metrics when the dashboard is first used by the user (i.e., when the index endpoint is called). https://anyscaleteam.slack.com/archives/C041GLHUXP0/p1669938709858869

**Review by ericl — APPROVED** [review]

File: dashboard/http_server_head.py [hunk 2]
```diff
@@ -101,6 +102,18 @@ def __init__(
 
     @routes.get("/")
     async def get_index(self, req) -> aiohttp.web.FileResponse:
+        try:
+            # This API will be no-op after the first report.
+            # Note: We always record the usage, but it is not reported
+            # if the usage stats is disabled.
+            record_extra_usage_tag(TagKey.DASHBOARD_USED, "True")
+            logger.info("pinged")
```
Comment (ericl) [inline dashboard/http_server_head.py:110]:
```suggestion
```

**Review by alanwguo — APPROVED** [review]

File: dashboard/http_server_head.py (hunk 2 above)
Comment (rkooo567) [reply dashboard/http_server_head.py:110]: oops.. thanks for catching this

**Review by scv119 — APPROVED** [review]

**Pull request merged**